webAO/utils/fileExists.js

```diff
@@ -16,3 +21,22 @@ const fileExists = async (url) => new Promise((resolve, reject) => {
   xhr.send(null);
 });
 export default fileExists;
+
+/* Returns whether file exists.
+ * `url' is a URL, including the base URL for bw compat.
+ * If manifest is empty, check the old way.
+ * Otherwise, look it up in the manifest */
+const fileExistsManifest = async (url) =>
+      new Promise((resolve, reject) => {
+          if(client.manifest.length == 0) {
+              resolve(fileExists(url));
+              return;
+          }
+          const c_url = encodeURI(canonicalizePath(decodeURI(url.slice(AO_HOST.length))));
```

Comment:
In general, JS uses camel case, so it may be a good idea to stick with that schema. Also calling it the full canonical. 
Kinda nitpicking that since `c` has to be slightly interpreted